src/pip/_internal/operations/prepare.py

```diff
@@ -376,8 +376,7 @@ def prepare_linked_requirement(
 
         # TODO: Breakup into smaller functions
         if link.scheme == 'file':
-            path = link.file_path
-            logger.info('Processing %s', display_path(path))
+            logger.info('Processing %s (cached)', req.name)
```

Comment:
Would this not display "Processing foo (cached)" if I ran `pip install ./foo-1.0-py2.py3-none-any.whl`? I think you'd need to distinguish cached and explicitly specified wheels here.